doc/dev-guide/src/linting.md

```diff
@@ -0,0 +1,33 @@
+# Linting
+
+## Manual linting
+
+When checking the codebase with [`clippy`](https://doc.rust-lang.org/stable/clippy/index.html), it is recommended to use:
+
+```console
+$ cargo +beta clippy --all --all-targets --all-features -- -D warnings
```

Comment:
I think we're now using `cargo clippy`, that is, not using the clippy from beta.

Reply:
@djc Yup I somehow missed this one... My bad.
@vpochapuis After fixing this one this PR should be good to go.

Reply:
Ok!

I have also made related changes in `doc/dev-guide/src/coding-standards.md`